kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/DomainMetadataSuite.scala

```diff
@@ -779,105 +788,15 @@ class DomainMetadataSuite extends DeltaTableWriteSuiteBase with ParquetSuiteBase
-    var txnBuilder = createWriteTxnBuilder(Table.forPath(engine, tablePath))
-      .withDomainMetadata("foo", "configuration string")
-    if (isNewTable) {
-      txnBuilder = txnBuilder.withSchema(engine, testSchema)
-    }
-    if (tableProperties.nonEmpty) {
-      txnBuilder = txnBuilder.withTableProperties(engine, tableProperties.get.asJava)
-    }
-    txnBuilder.build(engine).commit(engine, emptyIterable())
-    // Check the feature is now supported
-    verifyDomainMetadataFeatureSupport(engine, tablePath)
-  }
-
-  test("automatically enable DomainMetadata when using withDomainMetadata - new table") {
-    // New table using the withDomainMetadata API
-    withTempDirAndEngine { (tablePath, engine) =>
-      verifyAutomaticUpgrade(engine, tablePath, isNewTable = true)
-    }
-  }
-
-  test("automatically enable DomainMetadata when using withDomainMetadata - existing " +
-    "table with legacy protocol") {
-    withTempDirAndEngine { (tablePath, engine) =>
-      // Create table with legacy protocol
-      createTxn(tablePath = tablePath, isNewTable = true, schema = testSchema, partCols = Seq())
-        .commit(engine, emptyIterable())
-      verifyAutomaticUpgrade(engine, tablePath)
-    }
-  }
-
-  test("automatically enable DomainMetadata when using withDomainMetadata - existing " +
-    "table with TF protocol") {
-    withTempDirAndEngine { (tablePath, engine) =>
-      // Create table with table feature protocol
-      createTxn(
-        tablePath = tablePath,
-        isNewTable = true,
-        schema = testSchema,
-        partCols = Seq(),
-        // Enable inCommitTimestamps to bump the protocol
-        tableProperties = Map(TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED.getKey -> "true"))
-        .commit(engine, emptyIterable())
-      verifyAutomaticUpgrade(engine, tablePath)
-    }
-  }
-
-  test("automatically enable DomainMetadata when using withDomainMetadata - existing " +
-    "table, upgrade two TF in same txn") {
-    withTempDirAndEngine { (tablePath, engine) =>
-      // Create table with legacy protocol
-      createTxn(tablePath = tablePath, isNewTable = true, schema = testSchema, partCols = Seq())
-        .commit(engine, emptyIterable())
-      verifyAutomaticUpgrade(
-        engine,
-        tablePath,
-        // Enable inCommitTimestamps as well
-        tableProperties = Some(Map(TableConfig.IN_COMMIT_TIMESTAMPS_ENABLED.getKey -> "true")))
-    }
-  }
-
   test("removing a domain on a table without DomainMetadata support") {
     withTempDirAndEngine { (tablePath, engine) =>
       // Create table with legacy protocol
       createTxn(tablePath = tablePath, isNewTable = true, schema = testSchema, partCols = Seq())
         .commit(engine, emptyIterable())
-      intercept[DomainDoesNotExistException] {
-        createWriteTxnBuilder(Table.forPath(engine, tablePath))
-          .withDomainMetadataRemoved("foo")
+      intercept[IllegalStateException] {
```

Comment:
Hmm I wonder if this should be an `IllegalStateException`?

Reply:
Also curious we chose to add `withDomainMetadataSupported` to the txnBuilder instead of making it automatic when using the APIs. Why is this? Possibly because we only want to upgrade the protocol in one place?

Reply:
Because withDomainMetadataSupported is on the `TransactionBulider`, whereas addDomainMetadata is on the `Transaction`.   We need to know in the builder_whether we support DM, so we can't wait till transaction to enable the feature support.

Reply:
> Hmm I wonder if this should be an IllegalStateException?

regarding this, any other error type you prefer?

Reply:
> Because withDomainMetadataSupported is on the TransactionBulider, whereas addDomainMetadata is on the Transaction. We need to know in the builder_whether we support DM, so we can't wait till transaction to enable the feature support.

I think essentially this comes down to only upgrading the protocol in the builder. In theory we could update the protocol in txn as well if needed (no concrete reason why we cannot). But I'm fine with this for now we can always revisit this later when we revisit our write APIs.